### ISSUE TITLE
multi-group deactivation prevents ancombc from running

### ISSUE THREAD

**Issue by jorondo1** [FrederickHuangLin/ANCOMBC#95]
Hi,
The following command:
```
anc <– ancombc(phyloseq=psSalivaSM, 
  p_adj_method="hochberg", 
  prv_cut=0.1, 
  formula="treatDay",
  group="treatDay", 
  alpha=0.05, 
  global=F, struc_zero=T, neg_lb=F, tol=1e-5, max_iter=100, conserve=F)
```
yields the following error whether or not `global=F`, and does not run. Any idea why?
`Error: (converted from warning) The group variable has < 3 categories
The multi-group comparison will be deactivated (global = FALSE)
`

running version 1.6.2

**Comment by jorondo1** [comment]
Some precision : I created a mock group variable with 3 levels and the command works... However, I am trying to run the non-global test on variables with 2 levels and can't figure it out. Thanks in advance for your help!

**Comment by FrederickHuangLin** [comment]
Hi @jorondo1,

`group` is only applicable for multi-group comparisons. If there are only two groups, you can set `group = NULL`, that should solve the problem. Let me know how it works.

Best,
Huang

**Comment by jorondo1** [comment]
Thanks for your reply, that works. However, I'm not sure I understand why that prevents structural zero detection?

**Comment by FrederickHuangLin** [comment]
Hi @jorondo1,

I am aware that many people are confused about the definition of structural zeros, so the following clarifications have been added to the new ANCOMBC release

> A taxon is considered to have structural zeros in some (>=1) groups if it is completely (or nearly completely) missing in these groups. For instance, suppose there are three groups: g1, g2, and g3. If the counts of taxon A in g1 are 0, but they are nonzero in g2 and g3, then taxon A will be considered to contain structural zeros in g1. In this example, taxon A is declared to be differentially abundant between g1 and g2, g1 and g3, and consequently, it is globally differentially abundant with respect to this group variable. Such taxa are not further analyzed using ANCOM-BC2, but the results are summarized in the overall summary.

Therefore, structural zeros are only meaningful where there are "groups".

In the next release, I will update the `group` argument for better clarification. Thanks for your feedback!

Best,
Huang

**Comment by jorondo1** [comment]
Thanks a lot for the quick response, it is quite clear now.

**Comment by FrederickHuangLin** [comment]
Hi @jorondo1,

Just want to make sure that the following message you saw is a warning message, not an error message, right?

```
The group variable has < 3 categories 
The multi-group comparison will be deactivated (global = FALSE)
```

Seeing this message should not prevent you from getting the primary result, is it?

Best,
Huang

**Comment by jorondo1** [comment]
Hi @FrederickHuangLin,
I noticed that. The problem was what you proposed above, that if I used a group variable with fewer than 3 groups, which prevented me from running it. I still got that warning message when trying with a group specification, which I thought weird since global was already set manually to F. Now with group=NULL I don't get that message anymore.

Cheers and thanks for following up,
Jo
